### PR TITLE
Feat/lambdas

### PR DESCRIPTION
**Pull request by sergioescudero**
Lesson that try to explain how to use Lambdas.

### CODE REVIEW

File: docs/advanced/lambda_functions.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Lambda functions
+
+Lambda expressions are ideally used when we need to do something simple and are
+more interested in getting the job done quickly rather than formally naming the
+function.
+
+Lambda expressions are also known as anonymous functions.
+
+Lambda expressions in Python are a short way to declare small, anonymous
```
Comment (McLargo) [inline docs/advanced/lambda_functions.md:9]: this paragraph is kind of repeating the two paragraphs above, right?

File: docs/advanced/lambda_functions.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+functions (you don't need to provide a name for lambda functions).
+
+Lambda functions behave like normal functions declared with the def keyword.
+They are useful when you want to define a small function concisely.
+They can contain only one expression, so they are not best suited for functions
```
Comment (McLargo) [inline docs/advanced/lambda_functions.md:14]: They `must` contain... I am not aware of lambda with more than one expressions

File: docs/advanced/lambda_functions.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+## Use with functions like map(), max(), etc
+
+Functions like map() can be used with lambda functions. In that case, map() will
+the lambda function to every element of a given iterable (list, tuple, etc.).
+
+For example:
+
+::: src.advanced.lambda_functions.get_list_of_fields_from_a_list_dict
+    options:
+      show_root_heading: true
```
Comment (McLargo) [inline docs/advanced/lambda_functions.md:33]: show_root_heading is already at root level, you can remove it from here

File: src/advanced/lambda_functions/lambda_functions.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        field_to_be_extracted: field to be extracted.
+
+    Returns:
+        list[str]: list of the values extracted.
+    """
+    return map(lambda d: d[field_to_be_extracted], content)
+
+
+def sort_a_list_of_dict_by_a_field(
+    content_to_be_sorted: list[dict],
```
Comment (McLargo) [inline src/advanced/lambda_functions/lambda_functions.py:22]: `list[dict[str, str]]` ?

File: src/advanced/lambda_functions/lambda_functions.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+) -> list[str]:
+    """
+    Extract and return a given field from a list of dictionaries.
+
+    Parameters:
+        content: list of dictionaries.
+        field_to_be_extracted: field to be extracted.
+
+    Returns:
+        list[str]: list of the values extracted.
```
Comment (McLargo) [inline src/advanced/lambda_functions/lambda_functions.py:16]: `list[dict[str, str]]`?

File: src/advanced/lambda_functions/lambda_functions.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    Sort a list of dict by a given field.
+
+    Parameters:
+        content_to_be_sorted: list of dict to be sorted.
+        sorted_by: field to be used for sorting.
+        asc: if false, it is sort in descending order. By default, it is true,
+        ascending order.
+
+    Returns:
+        list[dict]: content sorted.
```
Comment (McLargo) [inline src/advanced/lambda_functions/lambda_functions.py:36]: `list[dict[str, str]]` ?

File: src/advanced/lambda_functions/lambda_functions.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    if asc:
+        return sorted(content_to_be_sorted, key=lambda d: (d[sorted_by]))
```
Comment (McLargo) [inline src/advanced/lambda_functions/lambda_functions.py:39]: can you do it in one line like ` sorted(content_to_be_sorted, key=lambda d: (d[sorted_by]), reverse=not asc)`

File: src/advanced/lambda_functions/lambda_functions.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        reverse=True,
+    )
+
+
+def filter_by_applying_function_to_elements(
+    func: Callable,
+    elements: list[Any],
+) -> list[Any]:
+    """
+    apply a function to a list of elements.
```
Comment (McLargo) [inline src/advanced/lambda_functions/lambda_functions.py:52]: Uppercase Apply

File: tests/advanced/lambda_functions/test_lambda_functions.py [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+def test_get_a_field_from_a_list_of_dict() -> None:
+    brands = get_list_of_fields_from_a_list_dict(cars, "brand")
+    expected_brands = ["Toyota", "Saab", "Volkswagen"]
+
+    assert all([a == b for a, b in zip(brands, expected_brands)])
+
+
+def test_get_list_dict_sorted_by_a_field() -> None:
+    cars = [
```
Comment (McLargo) [inline tests/advanced/lambda_functions/test_lambda_functions.py:22]: why not use `cars` defined above? as you are sorting, I guess that is the idea behind the test

File: src/advanced/lambda_functions/lambda_functions.py [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    if asc:
+        return sorted(content_to_be_sorted, key=lambda d: (d[sorted_by]))
+    return sorted(
+        content_to_be_sorted,
+        key=lambda d: d[sorted_by],
+        reverse=True,
+    )
+
+
+def filter_by_applying_function_to_elements(
```
Comment (McLargo) [inline src/advanced/lambda_functions/lambda_functions.py:47]: this is cool!!

File: tests/advanced/lambda_functions/test_lambda_functions.py [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+            a == b
+            for a, b in zip(
+                brands_sorted_dsc,
+                expected_brands_sorted_by_release_year_dsc,
+            )
+        ],
+    )
+
+
+def test_pass_a_lambda_function_as_argument() -> None:
```
Comment (McLargo) [inline tests/advanced/lambda_functions/test_lambda_functions.py:70]: if you want to split the test in small bytes. meaning one test to assert even numbers, one for odd numbers and one for start with a

**Review by McLargo — CHANGES_REQUESTED** [review]

File: src/advanced/lambda_functions/lambda_functions.py (hunk 5 above)
Comment (sergioescudero) [reply src/advanced/lambda_functions/lambda_functions.py:16]: it is fine. it returns a list of values

File: src/advanced/lambda_functions/lambda_functions.py (hunk 7 above)
Comment (sergioescudero) [reply src/advanced/lambda_functions/lambda_functions.py:39]: true. I make a lesson explaining the use of lambdas for doing everything in a line and I leave that....

File: tests/advanced/lambda_functions/test_lambda_functions.py (hunk 9 above)
Comment (sergioescudero) [reply tests/advanced/lambda_functions/test_lambda_functions.py:22]: oops!

Comment (McLargo) [reply tests/advanced/lambda_functions/test_lambda_functions.py:25]: still here!!

**Review by McLargo — APPROVED** [review]

**Pull request merged**